android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt

```diff
@@ -255,6 +259,21 @@ open class Node<V : RibView>(
         children.forEach { it.onPause() }
     }
 
+    internal fun startActivityForResult(intent: Intent, requestCode: Int, options: Bundle? = null) {
+        activityStarter.startActivityForResult(intent, generateRequestCode(requestCode), options)
+    }
+
+    fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean  =
+        if (requestCodeRegistry.resolveGroupId(requestCode) == ribId) {
+            interactor.onActivityResult(requestCodeRegistry.resolveRequestCode(requestCode), resultCode, data)
+            true
+
+        } else {
+            children.asReversed().any {
```

Comment:
why .asReversed()?
If it some kind of optimizations do we really need it?

Reply:
If you have a back stack that keeps children alive, those children are not attached to the screen, and pretty sure they were not the ones launching an Activity for result. So it's better to start at the end of the list with children that were created most recently. What do you think?

Reply:
In case of backstack it will work, but in case if you have 1 Parent RIB and many small RIBs and every of this small rib can startActivity, this optimization will not work.

But probably in most of the cases we will have backStack, so your point is valid.
Maybe we at can add comment about your idea of optimisation?

Reply:
Added comment.